src/main/java/com/google/sps/servlets/UserDataServlet.java

```diff
@@ -154,5 +230,36 @@ private String getStringParameter(HttpServletRequest request, String name, Strin
     String[] value = request.getParameterValues(name);
     return value == null ? defaultValue : value;
   }
+
+  /** Returns the request parameter (for booleans), or the default value if not specified */
+  private boolean getBooleanParameter(HttpServletRequest request, String name, boolean defaultValue) {
+    String value = request.getParameter(name);
+    if (value == null) return false;
+    else return value.equals("true");
+  }
+
+  /** Gets the blobkey of the image passed to the designated input tag in HTML */
+  public String getUploadedFileBlobKey(HttpServletRequest request, String formInputElementName) {
+    Map<String, List<BlobKey>> blobs = blobstore.getUploads(request);
+    List<BlobKey> blobKeys = blobs.get(formInputElementName);
+
+    // User submitted form without selecting a file, so we can't get a URL. (dev server)
+    if (blobKeys == null || blobKeys.isEmpty()) {
+      return null;
+    }
+
+    // Our form only contains a single file input, so get the first index.
+    BlobKey blobKey = blobKeys.get(0);
+
+    // User submitted form without selecting a file, so we can't get a URL. (live server)
+    if (!testing) {
+      BlobInfo blobInfo = new BlobInfoFactory().loadBlobInfo(blobKey);
+      if (blobInfo.getSize() == 0) {
+        blobstore.delete(blobKey);
+        return null;
+      }
+    }
```

Comment:
Why do we need to branch this behavior for testing?

Reply:
I couldn't figure out another way to "get around" this snippet of code. I couldn't figure out how to mock the `BlobInfo blobInfo = new BlobInfoFactory().loadBlobInfo(blobKey);` line. Do you have a suggestion on how I would test the method while including that snippet?

Reply:
Okay so it turns out that code snippet wasn't necessary, I removed it.